staging/src/k8s.io/legacy-cloud-providers/openstack/metadata.go

```diff
@@ -76,6 +100,10 @@ type Metadata struct {
 	AvailabilityZone string           `json:"availability_zone"`
 	Devices          []DeviceMetadata `json:"devices,omitempty"`
 	// .. and other fields we don't care about.  Expand as necessary.
+
+	// We have to use AWS compatible metadata for the next fields, because OpenStack doesn't
+	// provide this information.
+	ec2Metadata
```

Comment:
`Metadata` is (at least before this PR) a struct used for deserializing the json object read from openstack metadata.  Squishing `ec2Metadata` into that seems weird and in conflict with that goal - at some point there's going to be a clash of json keys between the two formats, and it is going to be very surprising.

Perhaps rename `Metadata` into `openstackMetadata` and then create a new `Metadata` that squishes `openstackMetadata` and `ec2Metadata` ?  Then we can deserialize into the more-specific `openstackMetadata` struct where required (eg `parseMetadata()`) and avoid conflicts.